File: Data Structures & Algorithms/two-integer-sum/submission-6.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> diffs = new();
        for (int i=0; i < nums.Length; ++i) {
            int diff = target - nums[i];
            if (diffs.ContainsKey(diff)) return [diffs[diff], i];
            else diffs[nums[i]] = i;
        }
        return [];
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-6.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l = 0, r = nums.size()-1, m, n = nums.size();
        while (l <= r) {
            m = (l + (r-l) / 2); //no overflow
            if (nums[m] <= nums[((m+1)%n+n)%n] && nums[m] <= nums[((m-1)%n+n)%n]) {
                return nums[m];
            } else if (nums[m] > nums[r]) {
                l = m+1;
            } else {
                r = m-1;
            }
        }
        return -1;
    }
};
